maven/lib/dependabot/maven/version_parser.rb

```diff
@@ -0,0 +1,137 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+require "strscan"
+require "dependabot/maven/token_bucket"
+
+# See https://maven.apache.org/pom.html#Version_Order_Specification for details
+#
+module Dependabot
+  module Maven
+    class VersionParser
+      extend T::Sig
+      extend T::Helpers
+
+      ALPHA = -5
+      BETA = -4
+      MILESTONE = -3
+      RC = -2
+      SNAPSHOT = -1
+      SP = "sp"
+
+      sig { params(version: T.nilable(String)).returns(TokenBucket) }
+      def self.parse(version)
+        raise ArgumentError, "Malformed version string #{version}" if version.nil?
+
+        new(version).parse
+      end
+
+      sig { params(version: String).void }
+      def initialize(version)
+        @version = version
+        @token_bucket = T.let(TokenBucket.new(tokens: []), T.nilable(TokenBucket))
+        @parse_result = T.let(@token_bucket, T.nilable(TokenBucket))
+        @scanner = T.let(StringScanner.new(version.downcase), StringScanner)
+      end
+
+      sig { returns(TokenBucket) }
+      def parse
+        parse_version(false)
+
+        raise ArgumentError, "Malformed version string #{version}" if parse_result.to_a.empty?
```

Comment:
ditto on the argumentError feedback here

Reply:
I've added a comment on that line. In this case version is not quite empty as in "" but more like version = ('.' | '-')